src/Core/Grid/Definition/Factory/WebserviceDefinitionFactory.php

```diff
@@ -0,0 +1,160 @@
+    {
+        return (new ColumnCollection())
+            ->add((new BulkActionColumn('bulk_action'))
+                ->setOptions([
+                    'bulk_field' => 'id_webservice_account'
+                ])
+            )
+            ->add((new DataColumn('key'))
+                ->setName($this->trans('Key', [], 'Admin.Advparameters.Feature'))
+                ->setOptions([
+                    'field' => 'key'
+                ])
+            )
+            ->add((new DataColumn('description'))
+                ->setName($this->trans('Key description', [], 'Admin.Advparameters.Feature'))
+                ->setOptions([
+                    'field' => 'description'
+                ])
+            )
+            ->add((new DataColumn('active'))
+                ->setName($this->trans('Enabled', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'active'
+                ])
+            )
+            ->add((new ActionColumn('actions'))
+                ->setName($this->trans('Actions', [], 'Admin.Global'))
+            );
+    }
+
+    protected function getFilters()
+    {
+        return (new FilterCollection())
+            ->add((new Filter('key', TextType::class))
+                ->setTypeOptions([
+                    'required' => false
+                ])
+                ->setAssociatedColumn('key')
+            )
+            ->add((new Filter('description', TextType::class))
+                ->setTypeOptions([
+                    'required' => false
+                ])
+                ->setAssociatedColumn('description')
+            )
+            ->add((new Filter('active', ChoiceType::class))
+                ->setTypeOptions([
+                    'required' => false,
+                    'choices' => $this->statusChoices->getChoices(),
+                    'choice_translation_domain' => false,
+                ])
+                ->setAssociatedColumn('active')
+            )
+            ->add((new Filter('actions', SearchAndResetType::class))
+                ->setTypeOptions([
+                    'attr' => [
+                        'data-url' => $this->resetActionUrl,
+                        'data-redirect' => $this->redirectionUrl
+                    ]
+                ])
+                ->setAssociatedColumn('actions')
+            );
+    }
+
+    protected function getGridActions()
+    {
+        return (new GridActionCollection())
+            ->add((new SubmitGridAction('add_webservice'))
+                ->setName($this->trans('Add new', [], 'Admin.Actions'))
+                ->setIcon('add')
+                ->setOptions([
+                    'submit_route' => 'admin_webservice', //todo: change to add webservice link
+                    'submit_method' => 'GET'
+                ])
+            )
+            ->add((new SimpleGridAction('common_refresh_list'))
+                ->setName($this->trans('Refresh list', [], 'Admin.Advparameters.Feature'))
```

Comment:
Actually, it should be only localized in the _Admin.Actions_ domains since it is a generic call-to-action, button or link that can be found on many pages of the back office. Don't you think so? It falls currently under both the _messages_ and the _Admin.Advparameters.Feature_ files...

Reply:
make sense, but we will lose all the related translations: is it worth?